app/services/search_service.rb

```diff
@@ -0,0 +1,208 @@
+  end
+  
+  # DEPRECATED
+  def type_ahead(id)
+    warn "[DEPRECATED] SearchService.type_ahead is deprecated.  Use the TypeaheadService methods instead."
+    matches = []
+
+    notes(id).select("title,type,nid,path").each do |match|
+      matches << "<i data-url='"+match.path+"' class='fa fa-file'></i> "+match.title
+    end
+
+    DrupalNode.limit(5)
+        .order("nid DESC")
+        .where('(type = "page" OR type = "place" OR type = "tool") AND node.status = 1 AND title LIKE ?', "%" + id + "%")
+        .select("title,type,nid,path").each do |match|
+      matches << "<i data-url='"+match.path+"' class='fa fa-"+match.icon+"'></i> "+match.title
+    end
+
+    maps(id).select("title,type,nid,path").each do |match|
+      matches << "<i data-url='"+match.path+"' class='fa fa-"+match.icon+"'></i> "+match.title
+    end
+
+    users(id).each do |match|
+      matches << "<i data-url='/profile/"+match.name+"' class='fa fa-user'></i> "+match.name
+    end
+
+    tags(id).each do |match|
+      matches << "<i data-url='/tag/"+match.name+"' class='fa fa-tag'></i> "+match.name
+    end
+
+    return matches
+  end
+
+  # Run a search in any of the associated systems for references that contain the search string
+  def textSearch_all(srchString)
+    sresult = DocList.new
+    unless srchString.nil? || srchString == 0
+      # notes
+      noteList = textSearch_notes(srchString)
+      sresult.addAll(noteList.items)
+      
+      # DrupalNode search
+      DrupalNode.limit(5)
+      .order("nid DESC")
+      .where('(type = "page" OR type = "place" OR type = "tool") AND node.status = 1 AND title LIKE ?', "%" + srchString + "%")
+      .select("title,type,nid,path").each do |match|
+        doc = DocResult.fromSearch(match.nid,match.icon,match.path,match.title,"",0)
+        sresult.addDoc(doc)
+      end
+      # User profiles
+      userList = textSearch_profiles(srchString)
+      sresult.addAll(userList.items)
+      
+      # Tags
+      tagList = textSearch_tags(srchString)
+      sresult.addAll(tagList.items)
+      # maps
+      mapList = textSearch_maps(srchString)
+      sresult.addAll(mapList.items)
+      # questions
+      qList = textSearch_questions(srchString)
+      sresult.addAll(qList.items)
+    end
+    return sresult
+  end
+
+  # Search profiles for matching text
+  def textSearch_profiles(srchString)
+    sresult = DocList.new
+    unless srchString.nil? || srchString == 0
+      # User profiles
+      users(srchString).each do |match|
+        doc = DocResult.fromSearch(0,"user","/profile/"+match.name,match.name,"",0)
+        sresult.addDoc(doc)
+      end
+    end
+    return sresult
+  end  
+
+  # Search notes for matching strings
+  def textSearch_notes(srchString)
+    sresult = DocList.new
+    unless srchString.nil? || srchString == 0
+      # notes
+      find_notes(srchString,25).each do |match|
+        doc = DocResult.fromSearch(match.nid,"file",match.path,match.title,match.body.split(/#+.+\n+/,5)[1],0)
```

Comment:
the last line "match.body.split(/#+.+\n+/,5)[1]" pulls out the first
paragraph of the .md data of the research note. Or, more precisely, it
extracts the first lines without the '#' markup. I've been trying to use
this to give the Google-like "preview' of the data

On Mon, Aug 22, 2016 at 6:25 PM, Jeffrey Warren <notifications@github.com>
wrote:

> In app/services/search_service.rb
> <https://github.com/publiclab/plots2/pull/652#discussion_r75770666>:
>
> > +      # User profiles
> > +      users(srchString).each do |match|
> > +        doc = DocResult.fromSearch(0,"user","/profile/"+match.name,match.name,"",0)
> > +        sresult.addDoc(doc)
> > +      end
> > +    end
> > +    return sresult
> > +  end
> > +
> > +  # Search notes for matching strings
> > +  def textSearch_notes(srchString)
> > +    sresult = DocList.new
> > +    unless srchString.nil? || srchString == 0
> > +      # notes
> > +      find_notes(srchString,25).each do |match|
> > +        doc = DocResult.fromSearch(match.nid,"file",match.path,match.title,match.body.split(/#+.+\n+/,5)[1],0)
>
> Ah, this one here? Wait, remind me what it does?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/652/files/9e9cd8bba78188107053014f91f27293bdb8bbef#r75770666>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AOOfYPm8ir-I4nMJAlqB32VvwVoo0ZI1ks5qiiHkgaJpZM4JebSg>
> .
>